[kiali.io] landing page revisions, condense the latest videos sections

jshaughn opened the pull request:
Fixes https://github.com/kiali/kiali/issues/3455

@xeviknal Can you take an intitial look at this an dlet me know what you think?   Minimally there are a couple of things that need to be addressed:
- The youtube kiali features playlist needs to be reverse date-sorted like the community demo playlist.
- If we were to keep the Hightower quote we'd likely need to get permission?

Changes:
- revise hero text to be more direct/less verbose and take up less height
- add another quote
- revise About text to be more current and less verbose
  - remove extraneous video link to help streamline page
- revise Features title
- mild revision to blog section wording
- consolidate vids into one section with:
  - 1 latest scrum demo vid
  - 1 getting started vid (this vid probably should be replaced)
  - 5 latest feature vids

xeviknal commented:
- I really like the idea of mixing all kinds of videos in one section!
- I also do like the change of the H1. However, I'd like to double check the keywords in it. But definitely more direct.
- I like the 4 sentences next to the H1. However, something visual bugs me. And definitely, I'd add h2 style into them for letting know Google that this is important for us.
- Is it possible to have h2/h3 in section titles? Perhaps it is not part of the issue though. H1 is for the main title of the page, not appropriate for section headlines.

Does this make sense?

jshaughn commented:
@xeviknal ,

> I really like the idea of mixing all kinds of videos in one section!

Glad to hear it, it reduces the page size and keeps things more current.  I'm not sure who has the youtube permissions to play with our content there.  Maybe @lucasponce?  We need to reverse the sort on the feature video playlist.

> I also do like the change of the H1. However, I'd like to double check the keywords in it. But definitely more direct.

I modified it slightly.

> I like the 4 sentences next to the H1. However, something visual bugs me. And definitely, I'd add h2 style into them for letting 
know Google that this is important for us.

I changed wording slightly, emphasized the key words with color, and limited the underlining to just the final statement.  I did not use h2/h3 style, it's too big.

> Is it possible to have h2/h3 in section titles? Perhaps it is not part of the issue though. H1 is for the main title of the page, not appropriate for section headlines.

I changed it to h3 consistently and made a couple of other small tweaks for consistency.  It looks better to me.

Let me know if you want to change anything, we can get together to maybe finalize these changes.  I will promote the PR.

lucasponce commented:
> Glad to hear it, it reduces the page size and keeps things more current. I'm not sure who has the youtube permissions to play with our content there. Maybe @lucasponce? We need to reverse the sort on the feature video playlist.

I have rights but I can give rights to another user I think @abonas has privileges to do that IIRC

xeviknal commented:
@jshaughn I've re-arranged the features playlist so the latest videos will be the first to be listed. To see this in the PR, you should re-build the PR. Not very very important though. It can be double-checked in a subsequent PR.

lucasponce commented:
A couple of minor comments from my side:
![image](https://user-images.githubusercontent.com/1662329/131846047-25efdaef-cc00-4132-8941-c63c173e023a.png)

I wonder if the "validate", "visualize", "troubleshoot" and "Manage" can use the same color palette of the site, this new color is a little strange (good think of using a palette is that you have 3-4 colors, and that's all you don't need to choose).

A second comment, I guess Kiali is mature and perhaps the cites from the users can be easily outdated:
![image](https://user-images.githubusercontent.com/1662329/131846299-37d413bb-888e-446a-9c18-26240cd55976.png)

So perhaps could be better to just rely these kind of comments on Twitter and not in the landing page, I think.

But those are more personal thoughts from my side.

xeviknal commented:
Agree with @lucasponce. I think your latest commit introduced this issue on the styles.

jshaughn commented:
@lucasponce @xeviknal , updated colors and removed quotes for Hero section.  Better?

jshaughn commented:
Adding @lucasponce as reviewer since @xeviknal has gone on Holiday.

abonas commented:
thanks @jshaughn for doing all those improvements.
From looking at the preview, the "manage" statement has underline as opposed to the other 3, not sure if this is on purpose? It looks a bit strange.
I'd also consider instead of the blue wording of "validate,visualiaze, etc." which looks a bit like a link (web links are often blue), but is not a link here, to perhaps make it black and bold instead.
![Screenshot at 2021-09-05 16-46-50](https://user-images.githubusercontent.com/6277245/132129326-99b41f21-2ce0-487f-9098-00718e208a73.png)

abonas commented:
I wonder if this white separator can be deleted or shrunk? This will minimize the scrolling and reveal faster some lower sections to users.
![Screenshot at 2021-09-05 16-58-54](https://user-images.githubusercontent.com/6277245/132129669-aeae9bb2-39e8-4fa8-bd7b-d23435b5cf5b.png)

abonas commented:
> > Glad to hear it, it reduces the page size and keeps things more current. I'm not sure who has the youtube permissions to play with our content there. Maybe @lucasponce? We need to reverse the sort on the feature video playlist.
> 
> I have rights but I can give rights to another user I think @abonas has privileges to do that IIRC

can you explain please again what's needed and where so I can have a look? on youtube? currently the sorting on the website of the videos seems random?

lucasponce commented on themes/kiali/layouts/partials/home/about-section.html:
```diff
@@ -1,19 +1,18 @@
-      Check out our demo videos
-    </a>
+    <p>
+    To learn more about Kiali (in under 10 minutes), watch this video...
```
I'd suggest to remove the video or just keep only the link.
It's pretty old and it doesn't invite to scroll down the page.

lucasponce commented on themes/kiali/layouts/partials/home/about-section.html:
```diff
@@ -1,19 +1,18 @@
-    service mesh by inferring traffic topology and displays the health of your
-    mesh. Kiali provides detailed metrics, powerful validation, Grafana access,
-    and strong integration for distributed tracing with Jaeger.
+    Kiali is a management console for Istio service mesh. It provides robust
```
Style comments are pretty subjective from all angles.
As it happens with presentations with long text, this is a little bit hard to read:
![image](https://user-images.githubusercontent.com/1662329/132227625-988dcd31-a691-419e-856d-7065511105a1.png)

I don't think it has to be comprehensive, just to invite to jump to the next area "Features" and from that to the documentation.

To me it's hard to read 5 lines of text with small font in a first page.

jshaughn replied:
For me this is OK, leaving as-is.  I had already updated the text to hopefully be easier to read and more current.  With the other changes it's quite easy to go past this to the Features section, and for some people (like me) I like reading a short blurb.

lucasponce commented:
The underlying in the "Manage" looks like a clicked link:
![image](https://user-images.githubusercontent.com/1662329/132228165-69471643-78cf-49f8-9e4c-34396b35d9dd.png)

I'd put all in the same, if you like, with increased font

lucasponce commented:
I'd also add an "anchor" link to the events (in general, I'd repeat the same technique here, to put a white link to the relevant content)
![image](https://user-images.githubusercontent.com/1662329/132228494-f120de27-9f5f-4cf7-a294-a091cb4f52a8.png)

lucasponce commented:
In general, I see that at the end we are personalizing the template and we should try to avoid that.
Most of the suggestions that we are proposing here are about style and not content and I think we should choose a single template and work with that.

@jshaughn is in your side, take all my style suggestions more like optional, personally it's a matter of taste, to me that's the problem with the current system.

jshaughn commented:
> In general, I see that at the end we are personalizing the template

What template?  I'd be happy to use some fixed template.  I was only trying to respond to the existing issues.  I have zero desire to make this an endless task, which it it quickly becoming given the number of suggestions in the PR comments.  I will make one more pass and then defer to sub-prs or a group session.

lucasponce commented:
> What template? I'd be happy to use some fixed template. I was only trying to respond to the existing issues. I have zero desire to make this an endless task, which it it quickly becoming given the number of suggestions in the PR comments. I will make one more pass and then defer to sub-prs or a group session.

Sure, +1 on your comment.

I mean "template" all html and style that is not part of the "content".

I see more of these issues are focused on the "template" instead about the content, as it's taking a considerable amount of time for authors and reviewers.

So, yes, to me we should limit or differentiate issues that are related to "template" changes (style of the content), versus "content" itself.

I don't want to slow down this PR, I'm on the same page in that sense.

jshaughn commented:
@lucasponce @abonas 

> From looking at the preview, the "manage" statement has underline as opposed to the other 3, not sure if this is on purpose? It looks a bit strange.

It was on purpose.  The original idea here was to remove the currently verbose and dated question/answer text in favor of simple text emphasizing the general approach for using Kiali: validate, observe, troubleshoot.  The  underlined `Manage` was supposed to highlight that Kiali should be your management console.  I agree that the underline is too confusing as it looks like a clicked link.  And that the secondary blue color makes the other entries look like unclicked links.  Changes:

- color to black
- increased size of the keywords for impact
- added exclamation point after Mesh as a replacement forthe underline

> I wonder if this white separator can be deleted or shrunk? This will minimize the scrolling and reveal faster some lower sections to users.

I reduced the top and bottom section margins to reduce the gap size.

> can you explain please again what's needed and where so I can have a look? on youtube? currently the sorting on the website of the videos seems random?

The video order is directly linked to our YouTube playlist order.  The feature-video playlist was asc time-ordered but Xavi has changed to be desc ordered, so now the videos are ordered newest-first on this updated landing page.

> I'd suggest to remove the video or just keep only the link.

Fine by me, I changed it to a link.

OK, hopefully we are close now for this one.   Let's limit further changes to blocker-level objections.

lucasponce commented:
No blockers.
I'll try to add a followup PR for the Search + Icons in the landing page, I see some minor improvements.

But let's consolidate.
